src/promises/services.py

```diff
@@ -1,79 +1,52 @@
-                    CRONLOGGER.info(percent_of_common_subjects)
-                    # We want at least 4 subjects on both the case and the promise
-                    if len(case_subject_ids) > 4 and len(promise_subject_ids) > 4:
-                        CRONLOGGER.info('case and promise have more than 4 subjects each')
-                        # If the percent is 80% or higher we make a connection
-                        if percent_of_common_subjects >= 0.8:
-                            PromiseCase.objects.create(case=case, promise=promise)
-                            CRONLOGGER.info('PromiseCase connection created')
-                        # If the percent is between 50% and 79% we create a suggested connection
-                        elif percent_of_common_subjects >= 0.5:
-                            SuggestedPromiseCase.objects.create(case=case, promise=promise)
-                            CRONLOGGER.info('SuggestedPromiseBill connection created')
+    current_parliament = Parliament.objects.latest('created')
+    parliament_sessions_ids = ParliamentServices.parliament_session_ids_by_parliament(current_parliament)
+    current_promises = Promise.objects.filter(parliament=current_parliament.id) 
+    cases_in_sessions = Case.objects.filter(parliament_session__in=parliament_sessions_ids)
+
+    for case in cases_in_sessions:
+        case_subject_ids = CaseSubject.objects.filter(case=case).values_list('subject_id', flat=True)
+        
+        # We look at each promise within the same parliament
+        for promise in current_promises:
+            promise_subject_ids = PromiseSubject.objects.filter(promise=promise).values_list('subject_id', flat=True)
+            if len(promise_subject_ids) > 4:
+                # We want more than 4 subjects on the promise
+                case_subject_set = set(case_subject_ids)
+                promise_subject_set = set(promise_subject_ids)
+                # We use set intersection to determine the number of common subjects
+                number_of_common_subjects = len(case_subject_set.intersection(promise_subject_set))
+                number_of_common_subjects_float = float(number_of_common_subjects)
+                number_of_promise_subjects_float = float(len(promise_subject_ids))
+                percent_of_common_subjects = (number_of_common_subjects_float / number_of_promise_subjects_float) * 100
+                
+                # If the case and promise share a certain percentage of subjects
+                # then we create a connection between them
+                if percent_of_common_subjects >= 50:
+                    relationship_type = 'Suggested'
+                    if percent_of_common_subjects >= 80:
+                        relationship_type = 'Connected'
+                    
+                    # We check if the connection already exists
+                    casePromiseConnection = PromiseCase.objects.filter(case=case, promise=promise)
```

Comment:
huh, I'd think that Django had some sort of utility function for this

Reply:
there is the exists() function, but if it exists I still want to update a field, so its easier to move that into a variable, check if that exists and then update the field & save